Develop

sjacobs1 opened the pull request:
##Description
Merge -> Main

##Screenshots
<img width="342" alt="Screenshot 2024-05-15 at 6 28 49 AM" src="https://github.com/sjacobs1/Pet-Movie-Project-Repo/assets/162406787/aced72e8-1e8b-4d4a-a0f8-2cb93a60e5c1">
<img width="339" alt="Screenshot 2024-05-15 at 6 29 15 AM" src="https://github.com/sjacobs1/Pet-Movie-Project-Repo/assets/162406787/d69716d8-9c29-4ddc-983d-6b139dad5b3e">
<img width="343" alt="Screenshot 2024-05-15 at 6 29 30 AM" src="https://github.com/sjacobs1/Pet-Movie-Project-Repo/assets/162406787/1e1b5cc8-47f3-4d23-bd54-45ed6f8ff210">
<img width="347" alt="Screenshot 2024-05-15 at 6 29 41 AM" src="https://github.com/sjacobs1/Pet-Movie-Project-Repo/assets/162406787/1b438de5-2e28-49c2-b840-c64099aec0ea">


##Checklist
- [x] The code follows the project's coding standards. 
- [x]  100% Unit test coverage for Tested View Models 
- [x] All warnings Resolved.

sjacobs1 committed:
HomeScreen/NowPlaying/Commit

sjacobs1 committed:
MovieDetailsCommit

sjacobs1 committed:
PopularMoviesCommit

Added popular movies endpoint and made main view controller for entire home screen (all endpoints)

sjacobs1 committed:
TopRatedMovies commit

added top rated movies to home screen

sjacobs1 committed:
upcomingMovies commit

added upcoming movies to home screen

sjacobs1 committed:
SearchMovieFeatureCommit

Search for movie function added on separate screen

sjacobs1 committed:
TabBarCommit

Added tab bar to app screen for navigation between home and search pages

sjacobs1 committed:
UpcomingMoviesCollectionViewCommit

added collection view to homescreen for upcoming movies

sjacobs1 committed:
commit collectionview inside tableview

added collectionview inside table view for sections

sjacobs1 committed:
collectionview in table view commit

sjacobs1 committed:
login page commit

Added login page

sjacobs1 committed:
commit images working

sjacobs1 committed:
Save code

sjacobs1 committed:
movie details screen commit

movie details screen now displaying selected movie info from search and displayed movies

sjacobs1 committed:
changes made to PR

made changes where necessary based on PR comments

sjacobs1 committed:
changes made per request

added changes to code based on comments made by mentors

simekanidev committed:
Update MovieAppPetProject/MovieAppPetProject/SearchMovies/SearchMovies-Repository.swift

sjacobs1 committed:
Dependency injection and refactoring

sjacobs1 committed:
Merge branch 'feature/movie_details_card' of https://github.com/sjacobs1/Pet-Movie-Project-Repo into feature/movie_details_card

sjacobs1 committed:
Merge pull request #1 from sjacobs1/feature/movie_details_card

movie details screen commit

sjacobs1 committed:
Feature - Displaying all movies

##Description
Cleaned up view models, models and repos and added DI to get ready for displaying movie data. Too many file changes to get movies to display now, so I just made necessary changes where needed for now to keep files changed low.

## Screenshots/GIFs
no UI updates or changes to show

sjacobs1 committed:
Merge pull request #2 from sjacobs1/feature/display_all_movies_homepage

Feature - Displaying all movies

sjacobs1 committed:
Feature - Display all movie types on homescreen

##Description
all movie arrays (Now playing, Top rated, Popular and Upcoming) display on homepage

sjacobs1 committed:
Feature - Display Movies

Made title label outlet private

sjacobs1 committed:
Merge pull request #3 from sjacobs1/feature/homepage_UI

Feature - Display all movie types on home screen

sjacobs1 committed:
Feature - Login screen

##Description
Added login view model to handle login, and updated login UI

sjacobs1 committed:
placeholder colour change

changed placeholder color on username and password fields

sjacobs1 committed:
login view model and view controller refactor

took out completion in login function and adjusted accordingly

sjacobs1 committed:
segue update

put segue name in constants

sjacobs1 committed:
rename LoginViewModel

sjacobs1 committed:
Login Logic changed

Moved login logic to be handled by view model through delegate

sjacobs1 committed:
Update README.md

sjacobs1 committed:
refactor

Moved delegate to vie model and removed unnecessary 'with'

sjacobs1 committed:
Merge pull request #5 from sjacobs1/readMe_file

Update README.md

sjacobs1 committed:
spelling change

sjacobs1 committed:
Merge pull request #4 from sjacobs1/feature/login_page

Feature - Login screen

sjacobs1 committed:
CI/CD Pipeline

sjacobs1 committed:
login segue fixed

sjacobs1 committed:
added develop branch to pullRequest.yml file

sjacobs1 committed:
Merge pull request #7 from sjacobs1/feature/CI_CD

Feature/ci cd

sjacobs1 committed:
unit test for nowPlaying view model

Added unit test for the NowPlayingViewModel testable code

sjacobs1 committed:
spelling update

sjacobs1 committed:
gemFile fix

sjacobs1 committed:
Merge pull request #8 from sjacobs1/unit_test/movie_view_model

unit test for nowPlaying view model

sjacobs1 committed:
Feature - Movie details refactor

##Description

I had to refactor the movie details feature, since i was bypassing the API call at first. now, when users click on a movie, it makes an API call based on the movieID and fetched its details. movie details are displayed on screen.

sjacobs1 committed:
Computed variables syntax

sjacobs1 committed:
rename function and set private outlets

sjacobs1 committed:
Merge pull request #9 from sjacobs1/feature/movie_overview

Feature - Movie details refactor

sjacobs1 committed:
Feature- movie details screen update

added more movie details to the screen

sjacobs1 committed:
use constants and took out comment

sjacobs1 committed:
Merge pull request #10 from sjacobs1/feature/movie_details_UI

Feature- movie details screen update

sjacobs1 committed:
Feature - Add movies to watchlist

sjacobs1 committed:
code changes

small code format changes and used dependency injection for CoreData

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/AppDelegate.swift:
```diff
@@ -10,8 +10,6 @@ import UIKit
-
-
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         // Override point for customization after application launch.
```
can remove all these default comments in these files to remove warnings

sjacobs1 replied:
comments removed

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-Model.swift:
```diff
@@ -0,0 +1,28 @@
+    
+    enum CodingKeys: String, CodingKey {
+        case originalTitle = "original_title"
+        case overview
```
case overview, runtime, status
 Just to clean it up
 Then have the rest that you are changing

sjacobs1 replied:
done

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift:
```diff
@@ -0,0 +1,54 @@
+//
+
+import UIKit
+import SDWebImage
```
are you still using this?
if so, please remove this and look into another way of handling images

sjacobs1 replied:
package removed and images handled differently. pr raised for this 👍

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    private func setupViewModelObserver() {
+        movieDetailsViewModel.didUpdateDetails = { [weak self] in
```
this is a bit strange. What is it for?

sjacobs1 replied:
refactor removed this functionality

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift:
```diff
@@ -0,0 +1,74 @@
+
+    // MARK: - Variables
+    private let movieDetailsRepository: MovieDetailsRepositoryType
+    private var movieID: Int = 0
```
private var movieID = 0

sjacobs1 replied:
change made

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift:
```diff
@@ -0,0 +1,74 @@
+    private let movieDetailsRepository: MovieDetailsRepositoryType
+    private var movieID: Int = 0
+    var movieDetails: MovieDetails? {
+        didSet {
```
this is also a bit strange. Can you explain a bit more and i can see if there is a better way to do it

sjacobs1 replied:
refactor removed this functionality

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift:
```diff
@@ -0,0 +1,74 @@
+            case .success(let details):
+                self?.movieDetails = details
+            case .failure(let error):
+                print(error)
```
please look into error handling and not just print statements

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    // MARK: - Closure
+    var didUpdateDetails: (() -> Void)?
```
this is also a bit strange

sjacobs1 replied:
refactor removed this functionality

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/HomeScreen/HomeScreen-ViewController.swift:
```diff
@@ -0,0 +1,68 @@
+////
```
is this file ment ot be commented out? If so, please delete it

sjacobs1 replied:
deleted

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/HomeScreen/HomeScreenNowPlaying/NowPlaying-ViewModel.swift:
```diff
@@ -0,0 +1,40 @@
+                self?.nowPlayingMovies = nowPlaying.results ?? []
+                self?.delegate?.reloadView()
+            case .failure(let error):
+                print(error)
```
same here for error handling

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift:
```diff
@@ -0,0 +1,103 @@
+    @IBOutlet private weak var newTableView: UITableView!
+
+    // MARK: - Variables
+    private lazy var nowPlayingViewModel = NowPlayingViewModel(nowPlayingRepository: NowPlayingRepository(), delegate: self)
```
this is really coupled. This screen should never know about 4 view models..

sjacobs1 replied:
changed

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/HomeScreen/PosterCellCollectionViewCell.swift:
```diff
@@ -0,0 +1,27 @@
+    
+    
+  
+    @IBOutlet weak var posterCover: UIImageView!
```
private

sjacobs1 replied:
done

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Login/LoginViewModel.swift:
```diff
@@ -0,0 +1,39 @@
+
+class LoginViewModel {
+
+    // MARK: - Variables
```
Look into coredata or firebase for auth instead of hardcoded credentials

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/SearchMovies/SearchMovieTableViewCell.swift:
```diff
@@ -0,0 +1,36 @@
+
+class SearchMovieTableViewCell: UITableViewCell {
+
+    @IBOutlet weak var nibCell: UIView!
```
private

sjacobs1 replied:
change made

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/SearchMovies/SearchMovieTableViewCell.swift:
```diff
@@ -0,0 +1,36 @@
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+
+        // Configure the view for the selected state
```
remove comment

sjacobs1 replied:
removed

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Login/LoginViewController.swift:
```diff
@@ -0,0 +1,48 @@
+    @IBOutlet private weak var passwordInput: UITextField!
+
+    // MARK: - Variable
+    private var loginViewModel: LoginViewModel!
```
force unwrapping?
This shouldn't be allowed with swiftlint?
Please instantiate this viewModel properly
private lazy var loginViewModel = LoginViewModel(delegate: self)
Please remove it from viewDidLoad.

sjacobs1 replied:
changes made in login cleanup PR

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Login/LoginViewController.swift:
```diff
@@ -0,0 +1,48 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loginViewModel = LoginViewModel(delegate: self)
+        usernameInput.attributedPlaceholder = NSAttributedString(string: "Username", attributes: [NSAttributedString.Key.foregroundColor: UIColor.gray])
```
Please put these in a private function and call it in the viewDidLoad for neatness.
Maybe called setUpPlaceHolderText()

sjacobs1 replied:
changes made in login cleanup PR

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/HomeScreen/PosterCellCollectionViewCell.swift:
```diff
@@ -0,0 +1,27 @@
+
+class PosterCellCollectionViewCell: UICollectionViewCell {
+    
+    
```
all this extra space?
Please don't just ignore the swiftlint warnings.
It should be picking it up. Either it is not set up correctly or you are ignoring the warnings.
Please have 0 swiftlint warnings when going from develop to main

sjacobs1 replied:
cleaned up 👍

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift:
```diff
@@ -0,0 +1,103 @@
+}
+
+// MARK: - Delegate
+extension NewHomeViewController: NowPlayingViewModelDelegate, UpcomingMoviesViewModelDelegate, PopularMoviesViewModelDelegate, TopRatedMoviesViewModelDelegate {
```
why does it conform to o many delegates?
Just make 1 protocol that all of them conform to then put it here.
Please also don't call them delegates. They are protocols

sjacobs1 replied:
change made

sjacobs1 commented:
A lot of these files I haven't touched after installing the linter, sorry for the formatting. I will make sure all warning are resolved as I raise PRs for these changes

simekanidev commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-Repository.swift:
```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+protocol MovieDetailsRepositoryType {
+    func fetchMovieDetails(movieID: Int, completion: @escaping (Result<MovieDetails, CustomError>) -> Void)
```
use a type alias for "(Result<MovieDetails, CustomError>) -> Void"

sjacobs1 replied:
done

sjacobs1 committed:
refactor from pr comments

sjacobs1 committed:
bugfix - refactor Login code

sjacobs1 committed:
login refactor

sjacobs1 committed:
Merge pull request #13 from sjacobs1/bugfix/login_cleanup

Bugfix/login cleanup

sjacobs1 committed:
Merge branch 'develop' into feature/watchlist

sjacobs1 committed:
watchlist funstionality

added fetch items functions and delete function. watchlist displayed in a table view

sjacobs1 committed:
moved string to constants

sjacobs1 committed:
bugfix - segue to movie details

sjacobs1 committed:
Merge pull request #14 from sjacobs1/bugfix/search_refactor

bugfix - segue to movie details

kieran-woodrow commented:
i'm approving so i your code can go in asap, but please make all changes

sjacobs1 commented:
will do, I just want to get my core data feature into dev first then I'll be making cleanups today

sjacobs1 committed:
Merge pull request #11 from sjacobs1/feature/watchlist

Feature - Add movies to watchlist

sjacobs1 committed:
bugfix - poster cell code cleanup

sjacobs1 committed:
suggestion added

sjacobs1 committed:
Merge pull request #15 from sjacobs1/bugfix/poster_cell_cleanup

bugfix - poster cell code cleanup

sjacobs1 committed:
bugfix - removed SDWebImage package

sjacobs1 committed:
error on ci/cd run

trying to sort out the failed run on ci/cd jobs

sjacobs1 committed:
Merge pull request #16 from sjacobs1/bugfix/images_display_refactor

bugfix - removed SDWebImage package

sjacobs1 committed:
bugfix - refactor loading of UI data

sjacobs1 committed:
protocol move

moved protocol from view controller to view model

sjacobs1 committed:
refactor based on comments

sjacobs1 committed:
repository name change

sjacobs1 committed:
file name change

sjacobs1 committed:
Merge pull request #17 from sjacobs1/bugfix/movie_details_refactor

bugfix - refactor loading of UI data

sjacobs1 committed:
bugfix - model file cleanup

sjacobs1 committed:
bugfix - model cleanup

sjacobs1 committed:
Merge pull request #19 from sjacobs1/bugfix/model_cleanup

Bugfix/model cleanup

sjacobs1 committed:
bugfix - renamed files

sjacobs1 committed:
Merge pull request #20 from sjacobs1/bugfix/rename_files

bugfix - renamed files

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsRepository.swift:
```diff
@@ -0,0 +1,42 @@
+
+    // MARK: - Variable
+    private let networkManager = NetworkManager()
+    private let coreDataManager: CoreDataManager
```
if you're injecting coredata, then inject networkManager as well

sjacobs1 replied:
done

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift:
```diff
@@ -0,0 +1,61 @@
+
+    // MARK: - Variables
+    private lazy var movieDetailsViewModel = MovieDetailsViewModel(movieDetailsRepository: MovieDetailsRepository(coreDataManager: CoreDataManager()), delegate: self)
+    private lazy var imageLoader = APIImageLoader()
```
does this need to be lazy?

sjacobs1 replied:
nope

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewModel.swift:
```diff
@@ -0,0 +1,79 @@
+class MovieDetailsViewModel {
+
+    // MARK: - Variables
+    private let movieDetailsRepository: MovieDetailsRepositoryType
```
convention is to have public vars then private vars.
same as functions. All public funcs then private funcs 🤙🏼

sjacobs1 replied:
understood

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/HomeScreen/PosterCellCollectionViewCell.swift:
```diff
@@ -0,0 +1,28 @@
+    @IBOutlet private weak var titleLabel: UILabel!
+
+    // MARK: - Variable
+    private lazy var imageLoader = APIImageLoader()
```
same

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Login/LoginViewController.swift:
```diff
@@ -0,0 +1,51 @@
+        setUpPlaceHolderText()
+    }
+
+    @IBAction private func loginTapped(_ sender: Any) {
```
check your convention in other files. i think you had IBoutlets the IB actions together. Try stick to your conventions in the file :)

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/SearchMovies/SearchMoviesViewModel.swift:
```diff
@@ -0,0 +1,41 @@
+
+class SearchMoviesViewModel {
+
+    private let searchMoviesRepository: SearchMoviesRepositoryType?
```
public then private

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Watchlist/WatchlistTableViewCell.swift:
```diff
@@ -0,0 +1,27 @@
+    private var watchlistItem: WatchList?
+
+    // MARK: - IBOutlets
+    @IBOutlet private weak var savedMovieTitle: UILabel!
```
space and // MARK: - IBActions for actions

sjacobs1 replied:
will change

kieran-woodrow reviewed (dismissed):
Approving but have a lot of comments, please address them.
Code is looking good though. Well done🤙🏼

sjacobs1 committed:
bugfix - reload watchlist page

sjacobs1 committed:
small rename and refactor

sjacobs1 committed:
Merge pull request #21 from sjacobs1/bugfix/relaod_watchlist_content

bugfix - reload watchlist page

sjacobs1 committed:
feature - prevent duplicate saves

sjacobs1 committed:
Merge pull request #22 from sjacobs1/feature/prevent_duplicate_saves

feature - prevent duplicate saves

sjacobs1 committed:
bugfix - large titles

sjacobs1 committed:
Merge pull request #23 from sjacobs1/bugfix/large_titles

bugfix - large titles

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsRepository.swift:
```diff
@@ -40,3 +41,10 @@ class MovieDetailsRepository: MovieDetailsRepositoryType {
     }
 }
+
+// MARK: - MovieDetailsRespository Extension
```
I think i left a comment in the other PR that this doesn't need to be in an extension :)

sjacobs1 replied:
yea sorry bro, I'll move all the functions in extension as I touch them again 🙏

sjacobs1 committed:
bugfix - home screen UI change

sjacobs1 committed:
Clean up UI

added rounded edges to headings and posters

sjacobs1 committed:
Merge pull request #24 from sjacobs1/bugfix/home_screen_UI

bugfix - home screen UI change

sjacobs1 committed:
feature - details UI

sjacobs1 committed:
feature - tab bar navigation

sjacobs1 committed:
simulator test

sjacobs1 committed:
feature - tab bar navigation

sjacobs1 committed:
Merge pull request #26 from sjacobs1/feature/navigate_back

Feature/navigate back

sjacobs1 committed:
Feature - details screen UI update

sjacobs1 committed:
Merge branch 'develop' into bugfix/details_screen_UI

sjacobs1 committed:
feature - code tidy up

sjacobs1 committed:
scanfile change

sjacobs1 committed:
functions rename

sjacobs1 committed:
feature - light and dark mode

sjacobs1 committed:
remove casting in view controller

sjacobs1 committed:
Merge pull request #28 from sjacobs1/feature/light_and_dark_mode

feature - light and dark mode

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift:
```diff
@@ -0,0 +1,67 @@
+        releaseDateLabel.text = movieDetailsViewModel.releaseDate
+        runTime.text = movieDetailsViewModel.runTime.map { "\($0) mins" }
+        voteAverage.text = movieDetailsViewModel.voteAverage.map { "Rating: \(String(format: "%.1f", $0)) / 10" }
+        status.text = movieDetailsViewModel.status.map { "Release status: \($0)"}
```
status.text = movieDetailsViewModel.status.map { "Release status: \($0)" }

sjacobs1 replied:
will correct all these labels bracket spacing

sjacobs1 committed:
spacing cleanup

sjacobs1 committed:
Merge branch 'develop' into bugfix/details_screen_UI

sjacobs1 committed:
Merge pull request #27 from sjacobs1/bugfix/details_screen_UI

Bugfix/details screen UI

sjacobs1 committed:
bugfix - home screen cleanup

sjacobs1 committed:
remove deleted files

sjacobs1 committed:
code changes on comments.

sjacobs1 committed:
feature - search screen UI changes

sjacobs1 committed:
search bar functionality

sjacobs1 committed:
private function and outlet

sjacobs1 committed:
Merge pull request #30 from sjacobs1/feature/search_functionality

feature - search screen UI changes

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/SearchMovies/SearchMoviesViewModel.swift:
```diff
@@ -9,33 +9,73 @@ import Foundation
 
 protocol ViewModelDelegate: AnyObject {
```
ViewModelProtocol

sjacobs1 committed:
UI for logout

sjacobs1 committed:
feature - logout functionality

sjacobs1 committed:
Merge pull request #31 from sjacobs1/feature/logout

Feature/logout

sjacobs1 committed:
Merge branch 'develop' into bugfix/home_screen_code_cleanup

sjacobs1 committed:
Merge pull request #29 from sjacobs1/bugfix/home_screen_code_cleanup

bugfix - home screen cleanup

sjacobs1 committed:
featture - ui changes

sjacobs1 committed:
feature - keyboard login

sjacobs1 committed:
Merge pull request #32 from sjacobs1/feature/logo_and_launch_screen

featture - ui changes

sjacobs1 committed:
Merge pull request #33 from sjacobs1/feature/keyboard_for_login

feature - keyboard login

sjacobs1 committed:
feature- loading indicators

sjacobs1 committed:
feature - update loading indicators

sjacobs1 committed:
asset rename

sjacobs1 committed:
unit tests update

sjacobs1 committed:
bugfix - logout button change

sjacobs1 committed:
Merge pull request #34 from sjacobs1/feature/loading_indicators

feature- loading indicators

sjacobs1 committed:
bugfix- small code cleanups

sjacobs1 committed:
Merge pull request #35 from sjacobs1/bugfix/logout_button_change

bugfix - logout button change

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/AppDelegate.swift:
```diff
@@ -6,31 +6,45 @@
+    }()
+    
+    // MARK: - Core Data Saving support
+    func saveContext () {
```
this save function can be moved to your core data class

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Assets.xcassets/Me.imageset/Contents.json:
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "filename" : "IMG-20221015-WA0077.jpg",
```
please rename asset

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/CoreDataFiles/CoreDataManager.swift:
```diff
@@ -0,0 +1,67 @@
+        guard let context = context else { return }
+        context.delete(item)
+        do {
+            try context.save()
```
can just call saveContext () when that function is moved here

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/CoreDataFiles/CoreDataManager.swift:
```diff
@@ -0,0 +1,67 @@
+            let newItem = WatchList(context: context)
+            newItem.originalTitle = movieDetails.originalTitle
+            newItem.moviePoster = movieDetails.moviePoster
+            try context.save()
```
can just call saveContext () when that function is moved here

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsRepository.swift:
```diff
@@ -0,0 +1,50 @@
+class MovieDetailsRepository: MovieDetailsRepositoryType {
+
+    // MARK: - Variable
+    private let networkManager = NetworkManager()
```
can inject this as well like you did with core data

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift:
```diff
@@ -0,0 +1,73 @@
+//
+//  GetMovieDetails-ViewController.swift
```
file name different to class name

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsRepository.swift:
```diff
@@ -0,0 +1,50 @@
+//
+//  GetMovieDetails-Repository.swift
```
file name different to class name

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsModel.swift:
```diff
@@ -0,0 +1,26 @@
+//
+//  GetMovieDetails-Model.swift
```
file name different to class name

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewModel.swift:
```diff
@@ -0,0 +1,89 @@
+//
+//  GetMovieDetails-ViewModel.swift
```
file name different to class name

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewModel.swift:
```diff
@@ -0,0 +1,89 @@
+                self?.isMovieSaved = self?.movieDetailsRepository.isMovieSaved(movieTitle: details.originalTitle ?? "") ?? false
+                self?.delegate?.updateMovieDetailsUI()
+            case .failure(let error):
+                print(error)
```
remove any and all prints in the code please

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/HomeScreen/HomeScreen-ViewController.swift:
```diff
@@ -0,0 +1,68 @@
+////
+////  ViewController.swift
```
is this file meant to be commented out?

sjacobs1 replied:
should be gone now

sjacobs1 committed:
Merge pull request #36 from sjacobs1/bugfix/small_cleanups

bugfix- small code cleanups

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Login/LoginViewController.swift:
```diff
@@ -0,0 +1,72 @@
+
+    // MARK: - Outlets
+    @IBOutlet private weak var usernameInput: UITextField!
+    @IBOutlet private weak var passwordInput: UITextField!
```
newline and mark for IBAction

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Login/LoginViewModel.swift:
```diff
@@ -0,0 +1,43 @@
+//
+//  Login-ViewModel.swift
```
file name

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Watchlist/WatchlistScreenViewController.swift:
```diff
@@ -0,0 +1,74 @@
+class WatchlistScreenViewController: UIViewController {
+
+    // MARK: - IBOutlet
+    @IBOutlet private weak var watchlistTableView: UITableView!
```
new line with IBAction mark

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Watchlist/WatchlistScreenViewController.swift:
```diff
@@ -0,0 +1,74 @@
+        cell.configure(with: watchlistItem.originalTitle, item: watchlistItem)
+        cell.delegate = self
+        
+        cell.selectionStyle = .none
```
this caqn be done in the cell file

kieran-woodrow commented on MovieAppPetProject/MovieAppPetProject/Watchlist/WatchlistScreenViewController.swift:
```diff
@@ -0,0 +1,74 @@
+        let watchlistItem = watchlistViewModel.watchlistItems[indexPath.row]
+        cell.configure(with: watchlistItem.originalTitle, item: watchlistItem)
+        cell.delegate = self
+        
```
can remove new lines

kieran-woodrow reviewed (dismissed):
nice job. Small comments for small fixes. Can be done in a separate pr

sjacobs1 committed:
small changes

sjacobs1 committed:
error handling

sjacobs1 committed:
test

sjacobs1 committed:
bugfix - updated unit test

updated unit test to reach 100% coverage on movieViewModel

sjacobs1 committed:
trailing space clean

sjacobs1 committed:
Merge pull request #37 from sjacobs1/bugfix/code_clean

Bugfix/code clean
